src/components/text-area/TextArea.tsx

```diff
@@ -0,0 +1,273 @@
+import React, { forwardRef, useMemo, useRef, useState } from "react";
+import { Platform, TextInputProps, ViewStyle } from "react-native";
+
+import { Box, BoxProps, RNTextInput, TouchableProps } from "../../primitives";
+import { useTheme } from "../../theme";
+import {
+  cx,
+  generateBoxShadow,
+  RenderPropType,
+  runIfFn,
+  styleAdapter,
+  useOnFocus,
+  useOnHover,
+} from "../../utils";
+import { createComponent } from "../../utils/createComponent";
+import { composeEventHandlers } from "../../utils/mergeRefs";
+import { Spinner } from "../spinner";
+import { createIcon, Icon, TextAreaSuffix } from "..";
+
+export type TextAreaVariants = "outline" | "subtle" | "underline";
+export type TextAreaSizes = "sm" | "md" | "lg" | "xl";
+
+export interface TextAreaProps extends TextInputProps {
+  /**
+   * The TextArea Container Wrapper
+   */
+  textInputWrapperProps: BoxProps;
+  /**
+   * A Component Theme of How the input should look
+   * @default outline
+   */
+  variant: TextAreaVariants;
+  /**
+   * The Component Size of How large should the input be?
+   * @default md
+   */
+  size: TextAreaSizes;
+  /**
+   * Prefix for the TextArea component
+   */
+  prefix: RenderPropType;
+  /**
+   * Suffix for the TextArea component
+   */
+  suffix: RenderPropType;
+  /**
+   * Set to True, if the value of the input is invalid.
+   * @default false
+   */
+  invalid: boolean;
+  /**
+   * Set to True, if the input is loading.
+   * @default false
+   */
+  loading: boolean;
+  /**
+   * Suffix wrapper props extends Touchable Props
+   */
+  _suffixProps: TouchableProps;
+  /**
+   * A11y Label
+   */
+  accessibilityLabel: string;
+}
+
+interface DefaultTextAreaSpinnerProps extends Pick<TextAreaProps, "size"> {
+  spinnerStroke: ViewStyle;
+}
+
+export const DefaultTextAreaSpinner = (
+  state: DefaultTextAreaSpinnerProps,
+): JSX.Element => {
+  const { size, spinnerStroke } = state;
+  return <Spinner size={size !== "xl" ? "xs" : "md"} style={spinnerStroke} />;
+};
+
+const RNTextArea: React.FC<Partial<TextAreaProps>> = forwardRef<
+  typeof RNTextInput,
+  Partial<TextAreaProps>
+>((props, ref) => {
```

Comment:
@crstnmac  The `ref` over here does not get assigned to the Input component, I think we need to do a 
`mergeRefs([inputRef, ref])` , 

```
const localRef = useRef();
const inputRef = mergeRefs([ref, localRef])

// Assign this to the input component
<Input
...
...
ref={inputRef}
/>
```

I think with this, we can make the `ref.blur()` work.

Reply:
Did it work?